core/src/datayoga_core/job.py

```diff
@@ -90,21 +89,30 @@ def transform(self, data: List[Dict[str, Any]],
         # use marshal. faster than deepcopy
         transformed_data = marshal.loads(marshal.dumps(data)) if deepcopy else data
 
-        results = []
+        result = JobResult()
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+        # depending on context, you might debug or warning log that a running event loop wasn't found
+            loop = asyncio.get_event_loop()
+
         for step in self.steps:
             try:
-                transformed_data, results = asyncio.run(step.block.run(transformed_data))
+                processed, filtered, rejected = loop.run_until_complete(step.block.run(transformed_data))
+                result.filtered.extend(filtered)
+                result.rejected.extend(rejected)
+                transformed_data = [result.payload for result in processed]
             except ConnectionError as e:
                 # connection errors are thrown back to the caller to handle
                 raise e
             except Exception as e:
                 # other exceptions are rejected
                 logger.error(f"Error while transforming data: {e}")
-                utils.reject_records(data, f"{e}")
-                transformed_data, results = utils.produce_data_and_results(data)
-                break
+                result.rejected.extend([Result(Status.REJECTED,payload=row,message=f"{e}") for row in transformed_data])
+                return result
 
-        return transformed_data, results
+        result.processed.extend([Result(Status.SUCCESS,payload=row) for row in transformed_data])
```

Comment:
why do we need to return the Status as well while this is an array of processed?